### PR TITLE
AIP84: UI endpoint for config

### PR DESCRIPTION
**Pull request by vatsrahul1001**
**closes**: https://github.com/apache/airflow/issues/43166

### Description
In the old UI, we passed a lot of variables via global variables in FAB templates. This PR replaces that with `ui/config` so that UI can use it.

Response json

```
{
    "navbar_color": "#fff",
    "navbar_text_color": "#51504f",
    "navbar_hover_color": "#eee",
    "navbar_text_hover_color": "#51504f",
    "navbar_logo_text_color": "#51504f",
    "page_size": 100,
    "auto_refresh_interval": 3,
    "default_ui_timezone": "UTC",
    "hide_paused_dags_by_default": false,
    "instance_name": "Airflow",
    "instance_name_has_markup": false,
    "enable_swagger_ui": true,
    "require_confirmation_dag_change": false,
    "default_wrap": false,
    "warn_deployment_exposure": true,
    "audit_view_excluded_events": "",
    "audit_view_included_events": "",
    "is_k8s": false,
    "test_connection": "Disabled",
    "state_color_mapping": {
        "deferred": "mediumpurple",
        "failed": "red",
        "queued": "gray",
        "removed": "lightgrey",
        "restarting": "violet",
        "running": "lime",
        "scheduled": "tan",
        "skipped": "hotpink",
        "success": "green",
        "up_for_reschedule": "turquoise",
        "up_for_retry": "gold",
        "upstream_failed": "orange"
    }
}
```

### Testing

image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/api_fastapi/core_api/datamodels/ui/config.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    navbar_text_color: str
+    navbar_hover_color: str
+    navbar_text_hover_color: str
+    navbar_logo_text_color: str
+    page_size: int
+    auto_refresh_interval: int
+    default_ui_timezone: str
+    hide_paused_dags_by_default: bool
+    instance_name: str
+    show_trigger_form_if_no_params: bool
```
Comment (bbovenzi) [inline airflow/api_fastapi/core_api/datamodels/ui/config.py]: We just decided to drop this config: https://github.com/apache/airflow/issues/44256#issuecomment-2492357015

Comment (jscheffl) [reply airflow/api_fastapi/core_api/datamodels/ui/config.py]: Yeah, but this was 5min ago :-D

Comment (vatsrahul1001) [reply airflow/api_fastapi/core_api/datamodels/ui/config.py]: @bbovenzi removed `show_trigger_form_if_no_params `

File: airflow/api_fastapi/core_api/datamodels/ui/config.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    audit_view_included_events: str
+    is_k8s: bool
+    test_connection: str
+    state_color_mapping: dict
+
+
+class ConfigResponse(BaseModel):
+    """config serializer for responses."""
+
+    configs: Config
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/datamodels/ui/config.py]: `configs` makes me expect a `:list[Config]` type

```diff
- configs: Config
+ config: Config
```

File: airflow/api_fastapi/core_api/datamodels/ui/config.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    instance_name_has_markup: bool
+    enable_swagger_ui: bool
+    require_confirmation_dag_change: bool
+    default_wrap: bool
+    warn_deployment_exposure: bool
+    audit_view_excluded_events: str
+    audit_view_included_events: str
+    is_k8s: bool
+    test_connection: str
+    state_color_mapping: dict
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/datamodels/ui/config.py:44]: Shouldn't this be in the `settings` part of the response ? (IS_K8S_OR_K8SCELERY_EXECUTOR) too ?

(maybe I missed something and we decided to flatten the response, which i'm not against either)

If we do not have `settings` and `config` and everything is under flattened, we could just return a plain:
```json
{
"navbar_color": ...,
"other_key": ...
}
```

WIthout needing the extra level of nesting of `config` inside `config`

**Review by pierrejeambrun — COMMENTED** [review]
Looking good.

Just a few suggestions/questions.

Are we missing the `standalone_dag_processor` ?

File: airflow/api_fastapi/core_api/datamodels/ui/config.py (hunk 2 above)
Comment (vatsrahul1001) [reply airflow/api_fastapi/core_api/datamodels/ui/config.py]: Done

File: airflow/api_fastapi/core_api/datamodels/ui/config.py (hunk 3 above)
Comment (vatsrahul1001) [reply airflow/api_fastapi/core_api/datamodels/ui/config.py:44]: @bbovenzi suggested to remove `settings` and `config` and return flattened dict.  Agreed @pierrejeambrun removed config nesting.

**Comment by vatsrahul1001** [comment]
> Looking good.
> 
> Just a few suggestions/questions.
> 
> Are we missing the `standalone_dag_processor` ?

As per below inline from Brent we have to do this in [issue](https://github.com/apache/airflow/issues/44253)

```
Other config:
standalone_dag_processor [This should be checked in API, not the UI](https://github.com/apache/airflow/issues/44253)
```

**Review by pierrejeambrun — DISMISSED** [review]
Looks nice!

**Comment by pierrejeambrun** [comment]
> Are we missing the standalone_dag_processor ?

To me this feels like we just need to add an extra field `additional_config` with `conf.getboolean("scheduler", "standalone_dag_processor")` ? Maybe i'm missing something.

**Comment by vatsrahul1001** [comment]
> > Are we missing the standalone_dag_processor ?
> 
> At first glance, it feels like we just need to add an extra field to `additional_config` with `conf.getboolean("scheduler", "standalone_dag_processor")` ?
> 
> Maybe i'm missing something though.

@bbovenzi can you confirm on this?

**Review by bbovenzi — APPROVED** [review]
lgtm.

standalone dag processor should be checked at the API level instead.

**Review by pierrejeambrun — APPROVED** [review]
Nice thanks.

**Pull request merged**